Unit tests: Fix ledger test and update name

hewison-chris opened the pull request:
Improve the naming of a helper function in `base.d` and prevent spurious failures in `test.Ledger`.

hewison-chris committed:
Unit tests: `postAndEnsureTxInPool` rename

The function that takes a transaction and makes sure it's hash is in the
tx pools of the clients had the same name as the function which just
checks that the transaction hash is in the pools.

hewison-chris committed:
Unit test: LockHeight use of postAndEnsureTxInPool

hewison-chris committed:
`test.Ledger` Unit test: updated to ensure non empty block

Sometimes the test failed because the tx was not in the pools of all
clients and a block could be created without the intended tx.

Geod24 reviewed (approved):
LGTM, just need to fix the build.

linked0 commented on source/agora/test/Base.d:
```diff
@@ -1210,14 +1210,20 @@ public class TestAPIManager
+    }
+
+    /// Ditto
+    public void postAndEnsureTxInPool (Idxs) (in Transaction tx, Idxs indices)
```
How about reordering the parameters like other functions that have the `indices` as the first parameter.

hewison-chris replied:
Yes good idea.

hewison-chris replied:
I did this in the last commit so that the changes are easier to follow.

linked0 commented on source/agora/test/Base.d:
```diff
@@ -1210,14 +1210,20 @@ public class TestAPIManager
+    /// Post a transaction to clients and ensure it is accepted by them (put in the tx pool)
+    public void postAndEnsureTxInPool (Idxs) (in Transaction tx)
+    {
+        this.postAndEnsureTxInPool(tx, iota(this.clients.length);
```
```suggestion
        this.postAndEnsureTxInPool(tx, iota(this.clients.length));
```

hewison-chris replied:
Updated. Not the only bug though.

hewison-chris replied:
Fixed the bugs.

linked0 reviewed (approved):
LGTM, two suggestions.

hewison-chris committed:
Unit tests: reorder the params

The indices in similar functions are specified first so this brings
`postAndEnsureTxInPool` and `ensureTxInPool` in line with them. Also
renamed to `client_idxs` to be consistent.

kchulj reviewed (approved):
LGTM
